### ISSUE TITLE
skygo italy?

### ISSUE THREAD

**Issue by cipitaua** [mh0rst/skycast#1]
Hi,

I've tried skycast with the italian version of skygo, and it does not work. Do you think is possible to adapt skycast in order to apply also on skygo-IT?

thanks in advance,
Roberto

**Comment by mh0rst** [comment]
I made a new version, which should work for the tablet version, too. As a bonus, it tries to disable root check. Please download and install this file: http://www32.zippyshare.com/v/26713871/file.html
Please, do report back if it worked so I can publish the changes for everyone to use :)

**Comment by cipitaua** [comment]
Thank you for the new version. I've tried it but skygo still returns "error 4107" (HDMI detected).

Note that I'm using the DFox version of skygo, that works also on rooted devices, and should already enable HDMI output:
http://www25.zippyshare.com/v/89195735/file.html


Roberto

**Comment by cipitaua** [comment]
no way, the damn 4107 error is always there...


Roberto

**Comment by cipitaua** [comment]
Note that the free-to-view channel skynews24 is visible with HDMI, but the other channels (skysport, etc) aren't.

**Comment by mh0rst** [comment]
I fear that the HDMI blocking code is inside some native library which can't be changed easily with Xposed. Please download and install http://www45.zippyshare.com/v/17938883/file.html then reboot and try to open a blocked channel. After you received the 4107 error, go to the Xposed Installer and open the "Logs". You should see alot of  java.lang.Exception: "submit me" in there. Please save the log (in the app menu) and upload it somewhere, then pass me the link.
It should give me some information about the origin of the 4107 error.

**Comment by cipitaua** [comment]
Here it is!

**Comment by cipitaua** [comment]
log here:
http://dl.dropbox.com/u/10405722/xposed_error_20150116_162632.log

**Comment by mh0rst** [comment]
Hey, thanks for the log. I was right, the checks are in native code. This is the last try I can do with Xposed methods, this will intercept all errors from native code to the app. Please test and report back: http://www48.zippyshare.com/v/tLR3kO2R/file.html
If you still receive errors or the App crashes/stops working, I would have to write native code replacement functions for skycast. This is possible and I have seen some promising methods, but I need more experience and time to implement them. I am currently busy working on my bachelor thesis, so I can't provide an ETA.

**Comment by cipitaua** [comment]
now skygo gives a different error, says somethig like "error connecting to server, please change channel and try again"
The log seems empty:
http://dl.dropbox.com/u/10405722/xposed_error_20150116_200230.log

**Comment by cipitaua** [comment]
there's a firewall active; does skycast expect to connect directly to internet?

**Comment by mh0rst** [comment]
No, skycast itself doesn't use the internet. I assume this is due to the native code blocking the connection after trying to display the 4107 error, so unfortunately you'll have to wait. You can uninstall skycast for now, I will reply when I find a solution.
